File: Tatarstan/Bars.Gkh.Regions.Tatarstan/Entities/UtilityDebtor/SeizureOfProperty.cs
namespace Bars.Gkh.Regions.Tatarstan.Entities.UtilityDebtor
{
    using System;

    using Bars.Gkh.Enums;
    using Bars.Gkh.Modules.ClaimWork.Entities;

    /// <summary>
    /// Постановление о наложении ареста на имущество
    /// </summary>
    public class SeizureOfProperty : DocumentClw
    {
        //2 поля дублируют аналогичные в UtilityDebtorClaimWork

        /// <summary>
        /// Абонент
        /// </summary>
        public virtual string AccountOwner { get; set; }

        /// <summary>
        /// Тип абонента
        /// </summary>
        public virtual OwnerType OwnerType { get; set; }

        /// <summary>
        /// Реквизиты физ. лица
        /// </summary>
        public virtual string AccountOwnerBankDetails { get; set; }

        /// <summary>
        /// Год
        /// </summary>
        public virtual string Year { get; set; }

        /// <summary>
        /// Номер
        /// </summary>
        public virtual string Number { get; set; }

        /// <summary>
        /// Подномер
        /// </summary>
        public virtual string SubNumber { get; set; }

        /// <summary>
        /// Документ-основание
        /// </summary>
        public virtual string Document { get; set; }

        /// <summary>
        /// Дата вручения
        /// </summary>
        public virtual DateTime? DeliveryDate { get; set; }

        /// <summary>
        /// Аннулировано
        /// </summary>
        public virtual bool IsCanceled { get; set; }

        /// <summary>
        /// Причина аннулирования
        /// </summary>
        public virtual string CancelReason { get; set; }

        /// <summary>
        /// Должностное лицо
        /// </summary>
        public virtual string Official { get; set; }

        /// <summary>
        /// Местонахождение
        /// </summary>
        public virtual string Location { get; set; }

        /// <summary>
        /// Взыскатель
        /// </summary>
        public virtual string Creditor { get; set; }

        /// <summary>
        /// Реквизиты
        /// </summary>
        public virtual string BankDetails { get; set; }

        /// <summary>
        /// Орган, вынесший постановление (Подразделение ОСП)
        /// дублирует поле из ExecutoryProcess
        /// </summary>
        public virtual JurInstitution JurInstitution { get; set; }
    }
}

File: Tatarstan/Bars.Gkh.Regions.Tatarstan/libs/B4/model/utilityclaimwork/SeizureOfProperty.js
Ext.define('B4.model.utilityclaimwork.SeizureOfProperty', {
    extend: 'B4.base.Model',
    idProperty: 'Id',
    fields: [
        { name: 'Id' },
        { name: 'DocumentNumber' },
        { name: 'DocumentDate' },
        { name: 'Year' },
        { name: 'Number' },
        { name: 'SubNumber' },
        { name: 'Document' },
        { name: 'DeliveryDate' },
        { name: 'IsCanceled' },
        { name: 'CancelReason' },        
        { name: 'JurInstitution', defaultValue: null },
        { name: 'Official' },
        { name: 'Location' },     
        { name: 'Creditor' },
        { name: 'BankDetails' },
        { name: 'OwnerType', defaultValue: 20 },
        { name: 'AccountOwner' },
        { name: 'AccountOwnerBankDetails' }
    ],

    proxy: {
        type: 'b4proxy',
        controllerName: 'SeizureOfProperty'
    }
});